rules/aip0122/resource_id_output_only_test.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aip0122
+
+import (
+	"testing"
+
+	"github.com/googleapis/api-linter/rules/internal/testutils"
+)
+
+func TestResourceIdOutputOnly(t *testing.T) {
+	for _, test := range []struct {
+		name          string
+		FieldName     string
+		FieldBehavior string
+		problems      testutils.Problems
+	}{
+		{"ValidWithSuffix", "book_id", "[(google.api.field_behavior) = OUTPUT_ONLY]", testutils.Problems{}},
```

Comment:
do you think there's an issue where someone has a bespoke field with the `_id` suffix?

Or is there an AIP that prevents that?

Reply:
a more robust way of doing this is to make the check look for the resource singular (e.g. extract some suffix from the message or type), then use `{resource}_id` for the check. 

But there might be enough consistency to do that today (the AIPs don't enforce message/type consistency.

Reply:
Actually, I think something like this has to be done, since [aip-122 explicitly states](https://google.aip.dev/122#fields-representing-another-resource):

> If using the resource name is not possible and using the ID component alone is strictly necessary, the field should use an _id suffix (e.g. shelf_id).

which means it's very possible that users are using `{some_other_resource}_id` as a user-settable value today.

Reply:
> resource singular , then use `{resource}_id` for the check.
>
> But there might be enough consistency to do that today

Yeah I think things are close enough today to do this. While we'd like the message and resource type to be the same, if the linter consistently uses the `type` name/`singular` field, I think that is always the right answer (even if the message name is different).

 I'll add this logic.

Reply:
Done! Also did a bit of refactoring to consolidate `Resource.type` parsing logic.